shapeshift/web
# feat: add thor tradable asset script

## Description

Adds a script to dynamically generate the mapping from thor asset notation to CAIP-19.

`yarn generate:thor-asset-map`

We currently maintain a hardcoded object for this, but we should make this more dynamic.

We opted to make this a generation script, rather than a run-time conversion for safety, and to give us the chance to sanity check any changes/new assets.

An asset's trade status, however, _is_ still checked at runtime.

## Pull Request Type


## Issue (if applicable)

N/A, quality of life.

## Risk

Small. If something went wrong with the generation we'd simply not be able to trade to or from the assets via THORSwap, or use them in THORSavers.

This should be easy to check at runtime.

## Testing

- Check all thor-tradable assets are still apperaing as expected
- Check all THORSaver assets are still appering as expected

### Engineering

☝️

Note, we lost a few unsupported Etherum assets, and gained some, so the actual generated list varies slightly from the hardcoded one.

Also note, we do not currently parse BNB or native THOR assets, and so running the generation script will show these as un-parsable (this is fine).

### Operations

☝️

## Screenshots (if applicable)

<img width="525" alt="Screenshot 2023-06-07 at 7 28 48 pm" src="https://github.com/shapeshift/web/assets/97164662/b243ba0a-c68e-4a8b-8f34-3e6b58f0a809">
